Port to 11ty

genehack opened the pull request:
This is still work-in-progress (hence the draft status) but I wanted to share this initial starting point. 

None of the pages have been converted to Markdown yet, but I've pulled the common header/footer HTML out into a wrapping template. There are some basic instructions in the README on how to get things installed and how to start up a local web server to serve the built site. 

Proposed plan of work:
* ~convert everything else to use the wrapper~
* ~refactor some more common parts from the pages into conditional parts of the wrapper~
* extend the 11ty config to support processing markdown
* start converting pages to markdown

Wave me off if this doesn't look useful or like what you had in mind.

genehack committed:
Initial Node setup

genehack committed:
Set up 11ty infrastructure

genehack committed:
Port index and 404 page to wrapper

genehack committed:
Port about-us and code-of-conduct to wrapper

genehack committed:
Properly indent discord

genehack committed:
Add some minimal instructions

desplesda commented:
This is fantastic! Thank you so much for doing this!

Are there any resources or info you need from us as you work on this?

genehack commented:
> This is fantastic! Thank you so much for doing this!

Sure, no problem. I find this sort of thing kind of relaxing. :grin: 

> Are there any resources or info you need from us as you work on this?

Not at the moment, but I will holler if and when. For the moment, just gonna keep plugging away per the plan I laid out above.

genehack committed:
Port membership pages to wrapper

The membership page had some embedded Javascript powering the two
Stripe buttons. I moved that out into a dedicated JS file (under
assets) and added a conditional bit to the wrapper that allows a page
to say "hey, I have a bit of javascript to add!" (just in case there's
any more of this sort of thing I haven't run across yet).

The old site was loading the Stripe JS on every page. I _think_ it's
only needed on this page, to support that custom script. Accordingly,
I moved that <script> tag to be inside the "I have a custom script"
conditional. If/when that needs to get more complicated, the Stripe
bit probably should be controlled by a distinct boolean in the front
matter of the pages that need it.

genehack committed:
Port events to wrapper

genehack commented:
Okay, everything is using the wrapper for layout as of 8ed3310 -- now to see about getting some more of the common parts of the secondary pages refactored out...

genehack committed:
Add "secondary" page template and port secondary pages to use it.

genehack committed:
Add "event" template and convert "events" sub-pages to use it

genehack committed:
don't remember why this wasn't checked in
